src/Integration/LanguageGroup.cs

```diff
@@ -1,14 +1,15 @@
 //-----------------------------------------------------------------------
-// <copyright file="RuleSetGroup.cs" company="SonarSource SA and Microsoft Corporation">
+// <copyright file="LanguageGroup.cs" company="SonarSource SA and Microsoft Corporation">
 //   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
 //   Licensed under the MIT License. See License.txt in the project root for license information.
 // </copyright>
 //-----------------------------------------------------------------------
 
 namespace SonarLint.VisualStudio.Integration
 {
-    public enum RuleSetGroup
+    public enum LanguageGroup
```

Comment:
Since the applied QP can change we can't use 'QP key/name/identifer/itself' as the grouping mechanism since it can change in the middle and we will not be able to handle that gracefully.

Language as group - I considered that but decided that its too risky and unrelated change to do part of this work. So done the simplest thing which is just renaming and map it to language.